### DISCUSSION TITLE
main token not valid

### DISCUSSION THREAD

**Discussion by FormlessFang** [GiorgioBrux/nitro-sniper-enhanced#122] [Q&A] [answered]
When the sniper tries to redeem a code it gets this error, [Nitro Sniper] (ERROR) Tried to redeem code [5Z5vn6ZGJY57H5MF] but the main token is not valid. Is this a problem with my main account or the code? I am sniping with my alt.
![image](https://user-images.githubusercontent.com/74550856/126705655-4c261714-e5d4-4bf0-b3e1-68ff5395b1ad.png)
I am using Heroku.

**Comment by CheeseTheWeese** [comment]
this happend to me do. i just got the sniper and sniped 3 codes overnight but it says main token isnt valid. im also using heroku and sniping with my alt. please fix this

**Comment by CheeseTheWeese** [comment, marked as answer]
i fixed it. basically just get ur main token again and dont log out or login to any other devices.

**Reply by FormlessFang** [reply]
have you sniped any codes as of fixing it?

**Reply by CheeseTheWeese** [reply]
no. i dont know why but it always says "Already redeemed". My snipe time is really slow, idk why, so other snipers may have gotten to it first or the link was already used/invalid. 
![image](https://user-images.githubusercontent.com/88008654/127756248-b926a374-fe9a-4cf1-a0da-ec4059236bcb.png)

**Reply by FormlessFang** [reply]
man, i haven't even sniped any, i either get "already checked" or invalid

**Reply by GiorgioBrux** [reply]
> i fixed it. basically just get ur main token again and dont log out or login to any other devices.

You can login to other devices, just don't log out. If you use the discord client I would recommend logging from a browser, getting the token from there and then never using it again.
